### PR TITLE
[merged] Capture and show more human friendly messages.

### PR DESCRIPTION
**Pull request by ashcrow**
This change replaces HTTP output with a slightly more human friendly
version.

### CODE REVIEW

**Comment by cooktheryan** [comment]
@ashcrow I'm on pto all next week and won't be able to review until the 11th

**Comment by ashcrow** [comment]
@cooktheryan np. I'll reassign. Thanks for letting me know!

File: src/commctl/cli.py [hunk 1]
```diff
@@ -150,6 +150,42 @@ def __init__(self, conf):
...
+
+        :param resp: The response to look at.
+        :type resp: requests.Response
+        :param operation: The name of the operation. EX: get
+        :type operation: str
+        :rtype: varies
+        :raises: ClientError
+        """
+        # Handle 204 No Content as its own case
+        if resp.status_code == 204:
```
Comment (mbarnes) [inline src/commctl/cli.py]: Just for readability sake, maybe use named constants for specific status codes?

The `requests` module provides a set:

```
if resp.status_code == requests.codes.no_content:
```

File: src/commctl/cli.py [hunk 2]
```diff
@@ -150,6 +150,42 @@ def __init__(self, conf):
...
         self._con.auth = (conf['username'], conf['password'])
 
+    def _handle_status(self, resp, operation):
+        """
+        Generic handling of status responses.
+
+        :param resp: The response to look at.
+        :type resp: requests.Response
+        :param operation: The name of the operation. EX: get
+        :type operation: str
```
Comment (mbarnes) [inline src/commctl/cli.py]: You could eliminate the `operation` argument and use `resp.request.method`.  Slightly cleaner.

Comment (ashcrow) [inline src/commctl/cli.py]: :+1: will do

File: src/commctl/cli.py (hunk 1 above)
Comment (ashcrow) [inline src/commctl/cli.py]: Good idea.

**Comment by ashcrow** [comment]
:arrow_up_small:

**Comment by mbarnes** [comment]
LGTM!

@rh-atomic-bot r+

**Pull request closed without merging**